cypress/events/index.js

```diff
@@ -186,41 +188,85 @@ module.exports = function setupNodeEvents (on, config) {
     }
 
     const retry = async () => {
-      console.log(`Will retry in ${delay} milliseconds`)
+      log(`Will retry in ${delay} milliseconds`)
       await sleep(delay)
       await pluginInstalled(plugin, version, timeout - delay)
     }
 
     return new Promise((resolve) => {
       const pkgFilePath = pathToPackageFile(plugin)
-      console.log(`Waiting for ${pkgFilePath} to exist`)
+      log(`Waiting for ${pkgFilePath} to exist`)
       return existsFile(pkgFilePath, timeout).then(() => {
         if (version) {
           const packageContent = fs.readFileSync(pkgFilePath, 'utf8')
           const { version: currentVersion } = JSON.parse(packageContent)
-          console.log(`Current version of ${plugin} is @${currentVersion}`)
+          log(`Current version of ${plugin} is @${currentVersion}`)
           if (version === '@' + currentVersion) {
             resolve(makeSureCypressCanInterpretTheResult)
           } else {
             retry().then(() => resolve(makeSureCypressCanInterpretTheResult))
           }
         } else {
-          console.log('Skip version test')
+          log('Skip version test')
           resolve(makeSureCypressCanInterpretTheResult)
         }
       })
     })
   }
 
-  const restoreStarterFiles = () => {
-    const { projectFolder } = config.env
-    const appDir = path.join(projectFolder, 'app')
-    return fse.emptyDir(appDir)
-      .then(() => fse.copy(starterDir, projectFolder))
-      .then(() => waitUntilAppRestarts())
+  const backupStarterFiles = () => {
+    const projectDir = path.join(config.env.projectFolder)
+    const backupDir = path.join(config.env.tempFolder, 'backupStarterFiles')
+
+    // Define the filter function
+    const filter = (dir) => !dir.includes('node_modules')
+
+    return fse.emptyDir(backupDir)
+      // Copy the files using the filter
+      .then(() => fse.copy(projectDir, backupDir, { filter }))
+      // Do not back up the package json
+      .then(() => fse.remove(path.join(backupDir, 'package-lock.json')))
       .then(makeSureCypressCanInterpretTheResult)
   }
 
+  const restoreStarterFiles = async (retries = 0) => {
```

Comment:
I think this would be better as `remainingRetries` - particularly in case it gets used when calling the function later on.

Reply:
Changed as requested